### DISCUSSION TITLE
[Python SDK] <how to swap tokens in one wallet?>

### DISCUSSION THREAD

**Discussion by ManutdGTA** [zkSync-Community-Hub/zksync-developers#104] [SDKs] [answered]
### Environment

Mainnet

### Acknowledgement

- [X] I understand that issues are expected with the Go SDK as they are being updated.

### Issue Description

I have encountered problems on Swap Token. There are two Orders in the document. 
For example,now  I want to swap ETH to USDC now, but I don’t know how to get the second order。
by the way ,submitter is wrong, I changed 'wallet'

### Expected Behavior

I can swap token in one order in one wallet
```
order= await wallet.get_order('USDT', 'ETH', Fraction(1500, 1), RatioType.token, Decimal('10.0'))
tx = await wallet.swap((order), 'ETH')

```

### Code Example

```orderA = await walletA.get_order('USDT', 'ETH', Fraction(1500, 1), RatioType.token, Decimal('10.0'))
orderB = await walletB.get_order('ETH', 'USDT', Fraction(1, 1200), RatioType.token, Decimal('0.007'))
tx = await submitter.swap((orderA, orderB), 'ETH')
```

**Comment by ManutdGTA** [comment]
@uF4No @MexicanAce plz experts,I really need help.I have waited over a week,I have asked for help in dc and zigzag telegram,but noone answered

**Comment by danijelTxFusion** [comment, marked as answer]
[Atomic](https://docs.zksync.io/dev/swaps/#atomic-swaps) swap provides token swap within a single account. 

```python
orderA = await wallet.get_order('USDT', 'ETH', Fraction(1500, 1), RatioType.token, Decimal('10.0'))
orderB = await wallet.get_order('ETH', 'USDT', Fraction(1, 1200), RatioType.token, Decimal('0.007'))

swap = await wallet.swap((orderA, orderB), 'ETH')
```

Pay attention to following limitations ([docs](https://docs.zksync.io/dev/swaps/#submitting-a-swap)):
- orders have matching tokens: if `orderA` specifies `tokenA -> tokenB`, then `orderB` should specify `tokenB -> tokenA`
- ratios in orders are compatible: `1/orderB.ratio <= orderA.amount/orderB.amount <= orderA.ratio`
- if orders have recipients, their accounts already exist in zkSync

**Reply by ManutdGTA** [reply]
thx for reply.
I tried, but it showed

Rejection reason  Self-swap is not allowed
zkSync tx hash 0xd5352d8ca76ed020916ffeadca3da80f10facb50d0f3c6c9ccc27d58dd6e47a6

**Comment by threadaqged** [comment]
To get the second order for swapping tokens, you need to call the get_order function on the appropriate wallet object with the desired token pair and parameters. In your case, to swap from ETH to USDT, you should use the get_order function on walletB with the token pair 'ETH' to 'USDT', specifying the desired price ratio and amount. Once you have both orders, you can then proceed to execute the swap transaction using the swap function with both orders as parameters.

**Comment by bartlettdv** [comment]
Environment
Mainnet

Acknowledgement
 I understand that issues are expected with the Go SDK as they are being updated.
Issue Description
I have encountered problems on Swap Token. There are two Orders in the document.
For example,now I want to swap ETH to USDC now, but I don’t know how to get the second order。
by the way ,submitter is wrong, I changed 'wallet'

Expected Behavior
I can swap token in one order in one wallet

order= await wallet.get_order('USDT', 'ETH', Fraction(1500, 1), RatioType.token, Decimal('10.0'))
tx = await wallet.swap((order), 'ETH')

Code Example
orderB = await walletB.get_order('ETH', 'USDT', Fraction(1, 1200), RatioType.token, Decimal('0.007'))
tx = await submitter.swap((orderA, orderB), 'ETH')
Repo Link (Optional)
No response

**Comment by tathimustingjason** [comment]
To obtain the second order for swapping ETH to USDC, you can use the same method as getting the first order but reverse the token pair and adjust the parameters accordingly.

**Comment by seimabirthcan** [comment]
To swap tokens, you typically need to create separate orders for each side of the trade. In your example, you would first obtain an order to swap ETH to USDC using one wallet (walletA), and then obtain a second order to swap USDC back to ETH using another wallet (walletB). Finally, you can execute the swap transaction using both orders. Make sure to handle the orders correctly based on your specific exchange or trading protocol.

**Comment by dianocsemintrey** [comment]
The user is encountering problems with swapping tokens using the Go SDK on the zkSync Mainnet. The user has two orders in the document and wants to swap ETH to USDC using the second order, but is unsure how to retrieve it. The user also mentions that the submitter is wrong and has changed the wallet. The expected behavior is to be able to swap tokens in one order in one wallet. The user provides code examples using the get_order and swap functions. No repo link is provided.
